### PR TITLE
Decoupled entrances/mixed pools, excluding grottos

### PR DESCRIPTION
**Pull request by puddydoo**
Rather than decoupled entrances and mixed pools being their own settings, they are a dropdown for each pool. This makes it possible to decouple and/or mix some pools but not others. Grottos remain a toggle so that there is no option to decouple or mix them, because of their issues.

It is possible to select whether individual decoupled pools are decoupled unidirectionally or bidirectionally. (On roman's fork it would always be unidirectional unless pools were mixed in which case they would all be bidirectional)

### CODE REVIEW

**Comment by fenhl** [comment]
I don't think “Insanity” is a good name. Can it just be “Mixed Decoupled”?

**Comment by puddydoo** [comment]
Probably. I picked it based off ALTTPR

**Comment by Zannick** [comment]
We don't have decoupled/mixed pools on this fork yet, do we?

Also yes, prefer descriptive names when possible. "Insanity" is not descriptive.

**Comment by Roman971** [comment]
Both decoupled and mixed pools aren't quite ready to be included in this branch yet. There are still some outstanding issues that would have to be fixed before they can be included in stable builds of the rando. I might have a chance to work on some of these issues soon, but it will probably still take a while before it gets there.
I also strongly disagree with turning every setting into a dropdown instead of having global options for this, since it adds a ton of complexity and inconsistency for no real gain, but that's probably a discussion for another day.

**Comment by puddydoo** [comment]
Even though grotto shuffle isn't included in this, I would enjoy the setting of decoupling grottos but not other entrances, and dropdown would allow such.

File: SettingsList.py [hunk 1]
```diff
@@ -2667,36 +2668,77 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
+            The pool of interior entrances contains most Houses and
+            all Great Fairies, but excludes the entrances to Windmill,
+            Link's House, Temple of Time, and Kakariko Potion Shop.
+
+            'Shuffle': Shuffle interior entrances among each other.
+
+            'Mix': Shuffle interior entrances into the "Mixed" pool
+            together with any other entrance pools set to 'Mix'.
+
+            'Decouple': Shuffle interior entrances among each other,
```
Comment (castlez) [inline SettingsList.py:2693]: This isn't an available option in the choices list. I assume the intent was to remove this from the tooltip

Loving this branch, hope it gets merged!

**Comment by cjohnson57** [comment]
Sorry, even if it excludes the problem area, I don't think we should merge decoupled to main until Roman is comfortable drafting the PR himself.

**Pull request closed without merging**